docs/ops/sort/MatrixNMS_8.md

```diff
@@ -0,0 +1,167 @@
+  * **Required**: *No*
+
+* *output_type*
+
+  * **Description**: the tensor type of outputs `selected_indices` and `valid_outputs`.
+  * **Range of values**: `i64` or `i32`
+  * **Type**: `string`
+  * **Default value**: `i64`
+  * **Required**: *No*
+
+* *score_threshold*
+
+  * **Description**: minimum score to consider box for the processing.
+  * **Range of values**: a floating-point number
+  * **Type**: `float`
+  * **Default value**: `0`
+  * **Required**: *No*
+
+* *nms_top_k*
+
+  * **Description**: maximum number of boxes to be selected per class.
+  * **Range of values**: an integer
+  * **Type**: `int`
+  * **Default value**: `-1` meaning to keep all boxes
+  * **Required**: *No*
+
+* *keep_top_k*
+
+  * **Description**: maximum number of boxes to be selected per batch element.
+  * **Range of values**: an integer
+  * **Type**: `int`
+  * **Default value**: `-1` meaning to keep all boxes
+  * **Required**: *No*
+
+* *background_class*
+
+  * **Description**: the background class id.
+  * **Range of values**: an integer
+  * **Type**: `int`
+  * **Default value**: `-1` meaning to keep all classes
+  * **Required**: *No*
+
+* *decay_function*
+
+  * **Description**: decay function used to decay scores.
+  * **Range of values**: `gaussian`, `linear`
+  * **Type**: `string`
+  * **Default value**: `linear`
+  * **Required**: *No*
+
+* *gaussian_sigma*
+
+  * **Description**: gaussian_sigma parameter for gaussian decay_function.
+  * **Range of values**: a floating-point number
+  * **Type**: `float`
+  * **Default value**: `2.0`
+  * **Required**: *No*
+
+* *post_threshold*
+
+  * **Description**: threshold to filter out boxes with low confidence score after decaying.
+  * **Range of values**: a floating-point number
+  * **Type**: `float`
+  * **Default value**: `0`
+  * **Required**: *No*
+
+**Inputs**:
+
+*   **1**: `boxes` - tensor of type *T* and shape `[num_batches, num_boxes, 4]` with box coordinates. The box cooridnates are layout as `[xmin, ymin, xmax, ymax]`. **Required.**
+
+*   **2**: `scores` - tensor of type *T* and shape `[num_batches, num_classes, num_boxes]` with box scores. **Required.**
+
+**Outputs**:
+
+*   **1**: `selected_outputs` - tensor of type *T_THRESHOLDS* and shape `[number of selected boxes, 6]` containing the selected boxes with score and class as tuples `[class_id, box_score, xmin, ymin, xmax, ymax]`.
+
+*   **2**: `selected_indices` - tensor of type *T_IND* and shape `[number of selected boxes, 1]` the selected indices in the flattened input `boxes`, which are absolute values cross batches. Therefore possible valid values are in the range [0, num_batches * num_boxes -1].
+
+*   **3**: `selected_num` - 1D tensor of type *T_IND* and shape `[num_batches, ]` representing the number of selected boxes for each batch element.
+
+When there is no box selected, `selected_num` is filled with 0. `selected_outputs` is an empty tensor of shape (0, 6), and `selected_indices` is an empty tensor of shape (0, 1).
```

Comment:
The same comments as for the `MultiClassNMS` spec in https://github.com/openvinotoolkit/openvino/pull/5907

Reply:
@lazarevevgeny think twice. I am afraid I have to leave shape of selected_indices of [number of selected boxes, 1]. The reason is,

Talked with Baidu, they insist they need shape similar to Paddle.
We cannot solve this by insert Unsqueeze after NMS because of dynamic shape issue. I am afraid Unsqueeze can handle dynamic shape input. For now we have to make sure NMS is the last layer in the ngraph function.

Reply:
all other change requests are resolved except this shape issue.